lib/rouge/lexers/solidity.rb

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Solidity < RegexLexer
+      tag 'solidity'
+      filenames '*.sol', '*.solidity'
+      mimetypes 'text/x-solidity'
+
+      title "Solidity"
+      desc "Solidity, an Ethereum smart contract programming language"
+
+      # optional comment or whitespace
+      ws = %r((?:\s|//.*?\n|/[*].*?[*]/)+)
+      id = /[a-zA-Z_][a-zA-Z0-9_]*/
```

Comment:
This is the regex for an identifier, right? Then it should be
```suggestion
      id = /[a-zA-Z$_][a-zA-Z0-9$_]*/
```

Reply:
This is used in identifying a lot of things - keywords, constants, types... Down there:

https://github.com/rouge-ruby/rouge/pull/760/files#diff-319a3b79139c64edf651b2af31c0aab7R132

Where can the dollar sign be used in Solidity? I must've missed it.

Reply:
This is the regex for identifiers in Solidity and has been since the beginning :)

Reply:
Ah, OK. Perhaps `id` is misnamed here, the regex is used to identify keywords, constants, essentially known language constructs.

Reply:
ah, that's different then, thanks!

Reply:
It's a convention in Rouge lexers for `id` to be used for the regular expression that matches identifiers within the target language. This could be constants, variables, function names, etc. That'S the sense in which it's used here.

Reply:
Given the above, can `$` be used in identifiers in Solidity? For example, in variable names.

Reply:
Yes, $ is a valid part of an identifier, even as the first character.

Reply:
Hm, OK, I'll add `$` in, then. A local "visual test" shows that these kinds of identifiers are about the last thing left.

It's caught eventually as `token Name` - without discerning between `Name.Class`, `Name.Variable`, `Name.Function`, etc.; I hope this is good enough for now, and "perfect" can wait. ;)

Reply:
I think it's acceptable to leave it as-is for now. As you said, in the future someone can come in and improve it (the typical way to do this is to use groups in the `id` regex and then use these to assign a set of tokens using the `groups` method.

Reply:
Oh, one thing: could you add an example to the visual sample of an identifier that uses `$`?

Reply:
Yup! Added here: https://github.com/rouge-ruby/rouge/pull/760/commits/4ba1cf7f28c8efdc32d7b6f3ac59da443ac4f97c